dependencies.yaml

```diff
@@ -523,6 +523,7 @@ dependencies:
           - pytest-xdist
           - seaborn
           - *scikit_learn
+          - &xgboost xgboost>=2.1.3
```

Comment:
RAPIDS 24.12 only has XGBoost 2.1.2. Can we use that as the lower bound here if we are including this in 24.12?

```suggestion
          - &xgboost xgboost>=2.1.2
```

Also what is the oldest version of XGBoost this would still work with? Ideally that determines the lower bound. Having a larger range between oldest supported and current available will make this easier for users to install

Reply:
It would be 2.1.0, where XGBoost adopted UBJSON as the default serialization format.